library/alloc/src/raw_vec.rs

```diff
@@ -50,7 +60,7 @@ enum AllocInit {
 #[allow(missing_debug_implementations)]
 pub(crate) struct RawVec<T, A: Allocator = Global> {
     ptr: Unique<T>,
-    cap: usize,
+    cap: Cap,
```

Comment:
nit: consider a comment on this field for how a `usize::MAX` capacity is represented, since the `///` comment just above says that the `capacity` method will return `usize::MAX`, but that can't be stored in `Cap`.